components/higher-order/with-hooks/provider.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import createHooks from '@wordpress/hooks';
+import { Component } from '@wordpress/element';
+
+export default class HooksProvider extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.hooks = createHooks();
```

Comment:
Was it tested with the recently added recreate editor instance feature? `createHooks()` makes me think that there is going to be a new object created with a fresh state, so we might lose all registered hooks.

Reply:
2nd thing we should figure out pretty soon is how it relates to `createHooks()` call for blocks: https://github.com/WordPress/gutenberg/blob/master/blocks/hooks/index.js#L11.

Unless `createHooks` uses singletone pattern and it isn't an issue at all :)

Reply:
I don't think it does @gziolo - did you ever look into this?

Reply:
It doesn't use singletone . It's fine as it is, we took a completely different approach anyways :)